.github/workflows/build-package.yml

```diff
@@ -0,0 +1,134 @@
+        value: "agent-v3-packages-${{ jobs.vars.outputs.package_build_md5 }}"
+
+defaults:
+  run:
+    shell: bash
+
+concurrency:
+  group: ${{ github.ref_name }}-build
+  cancel-in-progress: true
+
+permissions:
+  contents: read
+
+jobs:
+  vars:
+    name: Setup variables
+    runs-on: ubuntu-22.04
+    outputs:
+      go_code_md5: ${{ steps.vars.outputs.go_code_md5 }}
+      package_rebuild: ${{ inputs.force_build || steps.package-cache.outputs.cache-hit != 'true' }}
+      packaging_md5: ${{ steps.vars.outputs.packaging_md5 }}
+      actions_md5: ${{ steps.vars.outputs.actions_md5 }}
+      package_build_md5: ${{ steps.vars.outputs.package_build_md5 }}
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@9bb56186c3b09b4f86b1c65136769dd318469633 # v4.1.2
+        with:
+          ref: ${{ inputs.branch }}
+
+      - name: Output Variables
+        id: vars
+        run: |
+          go_code_md5=$(find . -type f \( -name "*.go" -o -name go.mod -o -name go.sum -o -name -o -name "*.proto" -o -name "*.pgo" -o -name "buf.yaml" -o -name "data-plane-api.yaml" \) -not -path "./site*" -exec md5sum {} + | LC_ALL=C sort  | md5sum | awk '{ print $1 }')
+          echo "go_code_md5=$go_code_md5" >> $GITHUB_OUTPUT
+          packaging_md5=$(find scripts/packages nginx-agent.conf Makefile.packaging .nfpm.yaml -type f -exec md5sum {} + | LC_ALL=C sort  | md5sum | awk '{ print $1 }')
+          echo "packaging_md5=$packaging_md5" >> $GITHUB_OUTPUT
+          actions_md5=$(find .github -type f -exec md5sum {} + | LC_ALL=C sort  | md5sum | awk '{ print $1 }')
+          echo "actions_md5=$actions_md5" >> $GITHUB_OUTPUT
+          package_build_md5=$(echo "$go_code_md5 $packaging_md5 ${{ inputs.version }}-${{ inputs.build }}" | md5sum | awk '{ print $1 }')
+          echo "package_build_md5=$package_build_md5" >> $GITHUB_OUTPUT
+          cat $GITHUB_OUTPUT
+
+      - name: Fetch Cached Package Artifacts
+        id: package-cache
+        uses: actions/cache@0c45773b623bea8c8e75f6c82b208c3cf94ea4f9 # v4.0.2
+        with:
+          path: build
+          key: agent-v3-packages-${{ steps.vars.outputs.package_build_md5 }}
+        if: ${{ ! inputs.force_build }}
+      
+      - name: Display contents of release tarball
+        run: tar -tzf build/packages/nginx-agentv3.tar.gz
+        if: ${{ steps.package-cache.outputs.cache-hit == 'true' }}
+
+  build-packages:
+    if: ${{ needs.vars.outputs.package_rebuild == 'true' }}
+    name: Build packages
+    runs-on: ubuntu-22.04
+    needs: [vars]
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@9bb56186c3b09b4f86b1c65136769dd318469633 # v4.1.2
+        with:
+          ref: ${{ inputs.branch }}
+
+      - name: Setup package build environment
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y gpgv1 monkeysphere 
+          go install github.com/goreleaser/nfpm/v2/cmd/nfpm@v2.32.0
```

Comment:
instead of running this `go install` command, it would be better to run `make install-tools`. That way we can make sure that the nfpm version is kept up to date.

Reply:
I have updated to use the make target, thanks @dhurley